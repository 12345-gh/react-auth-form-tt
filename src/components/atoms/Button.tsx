import React from "react";
import styled from "styled-components";

import { mediaQueries } from "@/styles/mediaQueries";

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
    children: React.ReactNode;
}

const StyledButton = styled.button<ButtonProps>`
    width: 240px;
    height: 48px;
    padding: 15px 32px;
    gap: 10px;
    border-radius: 30px;
    opacity: 1;

    background: linear-gradient(110.46deg, #70c3ff 12.27%, #4b65ff 93.92%);
    color: #fff;
    border: none;
    font-size: 16px;
    font-weight: 700;
    line-height: 14px;
    text-align: center;
    cursor: pointer;
    transition: opacity 0.3s ease;

    &:hover {
        opacity: 0.7;
    }

    &:disabled {
        cursor: not-allowed;
        opacity: 0.5;
    }

    @media ${mediaQueries.xs} {
        max-width: 100%;
    }

    @media ${mediaQueries.sm} {
        max-width: 90%;
    }

    @media ${mediaQueries.md} {
        max-width: 70%;
    }

    @media ${mediaQueries.lg} {
        max-width: 50%;
    }
`;

export const Button: React.FC<ButtonProps> = ({ children, ...props }) => {
    return <StyledButton {...props}>{children}</StyledButton>;
};
