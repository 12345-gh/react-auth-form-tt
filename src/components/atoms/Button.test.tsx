import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, expect, it, vi } from "vitest";

import { Button } from "./Button";

describe("Button component", () => {
    it("matches the snapshot", () => {
        const { asFragment } = render(<Button>Click Me</Button>);
        expect(asFragment()).toMatchSnapshot();
    });
    it("renders the Button component with children", () => {
        render(<Button>Click Me</Button>);
        expect(screen.getByText("Click Me")).toBeInTheDocument();
    });

    it("responds to click events", async () => {
        const handleClick = vi.fn();
        render(<Button onClick={handleClick}>Click Me</Button>);
        await userEvent.click(screen.getByText("Click Me"));
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
