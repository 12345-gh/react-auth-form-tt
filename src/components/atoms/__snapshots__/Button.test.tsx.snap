// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-braxZu bIuojJ"
  >
    Click Me
  </button>
</DocumentFragment>
`;
